templates/agent/r10k.rb.erb

```diff
@@ -82,10 +96,11 @@ module MCollective
             cmd << 'cache' if action == 'cache'
             cmd << 'deploy' << 'environment' << '-p' if action == 'synchronize' or action == 'sync'
             if action == 'deploy'
-              cmd << 'deploy' << 'environment' << arg << '-p'
+              cmd << 'deploy' << 'environment' << arg
             elsif action == 'deploy_module'
               cmd << 'deploy' << 'module' << arg
             end
+            cmd += extra
```

Comment:
It is in it's present form, as 'extra' is empty in cache context - effectively a no-op.  I put it outside the conditional because it makes extending the other methods the same way in the future easy - anything can alter the way the the r10k call gets executed by just passing in an 'extra' of it's own.